src/app/components/top-compounds/top-compounds.component.ts

```diff
@@ -89,4 +96,19 @@ export class TopCompoundsComponent implements OnInit {
           reverse();
       }
     }
+
+    /* function to get top postive/negative annotatedPlateWellIds */
+    getTopAnnotatedPlateWellIds(annotatedPlatewellidsArray: AnnotatedPlatewellid[],
+       type: string): AnnotatedPlatewellid[] {
+      let subArray: AnnotatedPlatewellid[] = Array();
+
+      if (type === 'POSITIVE') {
```

Comment:
use a boolean instead for the "POSITIVE" flag.
change the property to `isPositive: boolean`.